compiler/rustc_errors/src/lib.rs

```diff
@@ -456,10 +459,18 @@ impl Handler {
                 emitted_diagnostics: Default::default(),
                 stashed_diagnostics: Default::default(),
                 future_breakage_diagnostics: Vec::new(),
+                quiet: false,
             }),
         }
     }
 
+    pub fn with_disabled_diagnostic<T, F: FnOnce() -> T>(&self, f: F) -> T {
+        self.inner.borrow_mut().quiet = true;
+        let ret = f();
+        self.inner.borrow_mut().quiet = false;
```

Comment:
I thought about it but wasn't sure if it was worth it because it could only be done from there. But two people thinking about it makes it worth it. I'll make the change.

Reply:
It mainly guards us from future changes introducing subtle bugs.